From 73dda107a404e80a108acef69718cd8ec4dcf397 Mon Sep 17 00:00:00 2001
From: berbascum <berbascum@ticv.cat>
Date: Sun, 1 Sep 2024 20:21:44 +0200
Subject: [PATCH] (compiler) warning: thermal_core.c:1698:8: snprintf size
 argument is too large

Fix: define vars to set the sizes

Warnings:
/buildd/sources/drivers/thermal/thermal_core.c:1698:8: warning: 'snprintf' size argument is too large; destination buffer has size 128, but size argument is 4096 [-Wfortify-source]
        ret = snprintf(boost_buf, PAGE_SIZE, buf);
              ^
/buildd/sources/drivers/thermal/thermal_core.c:1781:8: warning: 'snprintf' size argument is too large; destination buffer has size 128, but size argument is 4096 [-Wfortify-source]
       snprintf(board_sensor_temp, PAGE_SIZE, buf);
---
 drivers/thermal/thermal_core.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/thermal/thermal_core.c b/drivers/thermal/thermal_core.c
index 9d8dee2eb6f7..4e52234e82ac 100644
--- a/drivers/thermal/thermal_core.c
+++ b/drivers/thermal/thermal_core.c
@@ -12,6 +12,9 @@
 
 #define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
 
+#define BOOST_BUF_SIZE 128
+#define SENSOR_TEMP_SIZE 128
+
 #include <linux/module.h>
 #include <linux/device.h>
 #include <linux/err.h>
@@ -1695,7 +1698,7 @@ thermal_boost_store(struct device *dev,
 				      struct device_attribute *attr, const char *buf, size_t len)
 {
 	int ret;
-	ret = snprintf(boost_buf, PAGE_SIZE, buf);
+	ret = snprintf(boost_buf, BOOST_BUF_SIZE, "%s", buf);
 	return len;
 }
 
@@ -1778,7 +1781,7 @@ static ssize_t
 thermal_board_sensor_temp_store(struct device *dev,
 				struct device_attribute *attr, const char *buf, size_t len)
 {
-       snprintf(board_sensor_temp, PAGE_SIZE, buf);
+       snprintf(board_sensor_temp, SENSOR_TEMP_SIZE, "%s", buf);
 
        return len;
 }
-- 
2.39.2

